components/ui/Alert.tsx

```diff
@@ -25,41 +25,162 @@ import AlertErrorIcon from '../icons/AlertErrorIcon';
 import AlertSuccessIcon from '../icons/AlertSuccessIcon';
 import AlertWarnIcon from '../icons/AlertWarnIcon';
 import InfoIcon from '../icons/InfoIcon';
-import ThemeContext from '../ThemeContext';
-import './Alert.css';
+import { css, SonarUiCommonTheme, SonarUiCommonThemeConsumer, styled } from '../theme';
 
-type AlertDisplay = 'banner' | 'block' | 'inline';
+type AlertDisplay = 'banner' | 'inline' | 'block';
 type AlertVariant = 'error' | 'warning' | 'success' | 'info';
 
 export interface AlertProps {
   display?: AlertDisplay;
   variant: AlertVariant;
 }
 
+interface VariantInformation {
+  icon: JSX.Element;
+  color: string;
+  borderColor: string;
+  backGroundColor: string;
+}
+
+const StyledAlertIcon = styled.div<{ isBanner: boolean; variantInfo: VariantInformation }>`
+  ${props => css`
+    flex: 0 0 auto;
+    display: flex;
+    justify-content: center;
+    align-items: center;
```

Comment:
All these can be put outside of the prop-dependent body, and they will be part of a "base" class. The variable css will then be in other additional classes